Tests/SwiftProtobufTests/Test_JSON.swift

```diff
@@ -970,6 +970,60 @@ final class Test_JSON: XCTestCase, PBTestHelpers {
         assertJSONDecodeFails("{")
         assertJSONDecodeFails("}")
     }
+  
+  func testEncodeWithDefaultValues() {
```

Comment:
Looks like this maps to one of the C++ tests, maybe bring over a [few of the others](https://github.com/search?q=repo%3Aprotocolbuffers%2Fprotobuf+always_print_primitive_fields+path%3A*test.cc&type=code) also as they involve combining some of the options to help make sure there aren't indesired interactions?

Reply:
Added some more tests intending to mirror what c++ is doing